src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,3404 @@
+        numVars = 4;
+
+        fillVals[2] = &valsYBnds;
+        fillVals[3] = &valsXBnds;
+        fillVarIDs[2] = YBndsID;
+        fillVarIDs[3] = XBndsID;
+    }
+
+    // --- Fill horizontal coordinate values
+
+    // Calculate resolution for y and x and allocate value arrays
+    resY = (SW_Domain->max_y - SW_Domain->min_y) / numY;
+    resX = (SW_Domain->max_x - SW_Domain->min_x) / numX;
+
+    // Generate/fill the horizontal coordinate variable values
+    // These values are within a bounding box given by the user
+    // I.e., lat values are within [ymin, ymax] and lon are within [xmin, xmax]
+    // bounds are ordered consistently with the associated coordinate
+    for (gridNum = 0; gridNum < numX; gridNum++) {
+        valsX[gridNum] = SW_Domain->min_x + (gridNum + .5) * resX;
+
+        if (!domTypeIsSite) {
+            bndsIndex = gridNum * 2;
+            valsXBnds[bndsIndex] = SW_Domain->min_x + gridNum * resX;
+            valsXBnds[bndsIndex + 1] = SW_Domain->min_x + (gridNum + 1) * resX;
+        }
+    }
+
+    for (gridNum = 0; gridNum < numY; gridNum++) {
+        valsY[gridNum] = SW_Domain->min_y + (gridNum + .5) * resY;
+
+        if (!domTypeIsSite) {
+            bndsIndex = gridNum * 2;
+            valsYBnds[bndsIndex] = SW_Domain->min_y + gridNum * resY;
+            valsYBnds[bndsIndex + 1] = SW_Domain->min_y + (gridNum + 1) * resY;
+        }
+    }
+
+    // --- Write out all values to file
+    for (varNum = 0; varNum < numVars; varNum++) {
+        SW_NC_write_vals(
+            &fillVarIDs[varNum],
+            domFileID,
+            NULL,
+            *fillVals[varNum],
+            start,
+            fillCounts[varNum],
+            "double",
+            LogInfo
+        );
+
+        if (LogInfo->stopRun) {
+            goto freeMem; // Exit function prematurely due to error
+        }
+    }
+
+    // Fill domain variable with SUIDs
+    SW_NC_write_vals(
+        &domID,
+        domFileID,
+        NULL,
+        domVals,
+        start,
+        domCount,
+        "unsigned integer",
+        LogInfo
+    );
+
+// Free allocated memory
+freeMem:
+    dealloc_netCDF_domain_vars(
+        &valsY, &valsX, &valsYBnds, &valsXBnds, &domVals
+    );
+}
+
+/**
+@brief Fill the variable "domain" with it's attributes
+
+@param[in] domainVarName User-specified domain variable name
+@param[in,out] domID Identifier of the "domain"
+    (or another user-specified name) variable
+@param[in] domDims Set dimensions for the domain variable
+@param[in] domFileID Domain netCDF file identifier
+@param[in] nDomainDims Number of dimensions the domain variable will have
+@param[in] primCRSIsGeo Specifies if the current CRS type is geographic
+@param[in] domType Type of domain in which simulations are running
+    (gridcell/sites)
+@param[in] deflateLevel Level of deflation that will be used for the created
+variable
+@param[out] LogInfo Holds information on warnings and errors
+*/
+static void fill_domain_netCDF_domain(
```

Comment:
Did you want this to be the same for the `grid_mapping` attribute? More specifically, we have the value for a projected crs to be `"crs_projsc: x y crs_geogsc: <y name> <x name>"`. Did we want to keep the "x" and "y" in this?

Reply:
Is it safe to presume that we will use these names rather than the ones that will be provided in the input spreadsheet and if so we would simply check that `geo_*AxisName` is the same as the X and Y axis names from said spreadsheet?

I would also like to confirm the ordering of these names regardng the coordinate attributes. We currently list the grid map as `"crs_projsc: x y crs_geogsc: lat lon"`. If I'm not mistaken, these coordinates are opposite of each other (i.e., not `x y lon lat` or `y x lat lon`). Is the current ordering correct?

Reply:
For the first point: yes, we should be using the inputs from desc_nc.in for the domain because the spreadsheet does not provide all five; then check that what the spreadsheet provides is consistent with desc_nc.in (see section "check domain for consistent horizontal axes names" of the nc-input plan).

For the second point: the current order is correct (e.g., see example 5.13 in [CF-1.11](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11)), i.e., the order is x-axis y-axis for projected crs and latitude longitude for geographic crs (in our case of wgs84)